src/hardware/uart.cpp

```diff
@@ -0,0 +1,268 @@
+#include <cstring>
+
+#include "hardware/uart.h"
+
+/**
+ * @brief Construct a UART peripheral.
+ *
+ * @param port Which UART peripheral to communicate with. Refer
+ * to ST documentation for which UARTs can be used on which pins.
+ * @param baud Baud rate.  Most of the time, you will want 9600
+ * or 115200.
+ * @param gpio Which GPIO port both `tx` and `rx` belong to.
+ * @param tx The pin to transmit on.
+ * @param rx The pin to receive on.
+ */
+UART::UART(UART::Port port, uint32_t baud, GPIO::Port gpio, uint8_t tx,
```

Comment:
I had it setup like this originally, but apparently the same UART can use different pins.

Reply:
Yes it can, on the PCB we've kept it to use the same pin port but if we want to make this code as general as possible, we should update this. 

I think a solution is to just create some list of defines for this particular PCB configuration saying what pins numbers we're using. Then for future configurations we can create the same type of list, allowing us to reuse the same source code and not change pin numbers anywhere.

Reply:
I'm going to merge this in for now, but we should make a note to update it later